### DISCUSSION TITLE
Code injection via bookmark solution for packaged projects

### DISCUSSION THREAD

**Discussion by BlueDome77** [TurboWarp/extensions#1242] [Q&A]
I'm packaging osu! Full Remake into a desktop app for Mac and wanted to have the [Beatmap Browser](https://terrariamods-scratch.github.io/hacks/osu/) by [TerrariaMods](https://scratch.mit.edu/users/TerrariaMods/) working without using a bookmark to inject it as you can't do that inside Turbowarp Desktop obviously. I've tried many different ways including Packager Custom JS, JS Extension, and redirect blocks from the Network extension. The closest I've gotten is running the script with a JS Extension then seeing errors in the console:

![Screenshot 2024-01-14 at 4 40 13 PM](https://github.com/TurboWarp/packager/assets/135789443/480ebe3c-3778-4cb0-9dd8-e3d1c6c2355d)

Any help would be appreciated!

**Comment by GarboMuffin** [comment]
-> https://github.com/TurboWarp/packager/discussions/808
